### ISSUE TITLE
IntEnum as a value type

### ISSUE THREAD

**Issue by onlined** [googleapis/google-cloud-python#7563]
**For Datastore:**
I think that instances of `IntEnum` can be serialized as their integer values. Its implementation is easy, and it would allow to use enumeration types easily.

**Comment by tseaver** [comment]
@onlined Can you show an example of how you would use it?  If we convert an instance of `enum.IntEnum` to an int when storing it into an entity, that conversion will be lossy:  we won't get an `IntEnum` back after reading the entity later.

**Comment by onlined** [comment]
The example:
```py
class EmployeeType(IntEnum):
	PART_TIME = 0
	FULL_TIME = 1
	REMOTE = 2

# employee_entity is an instance of Entity
employee_entity['type'] = EmployeeType.PART_TIME.value # I don't want to enter .value
employee_entity.put()
```

**Comment by tseaver** [comment]
@onlined OK, but what do you expect to happen when fetching the entity later via `get()`?  If we convert the enum to its integer value implicitly, then you will just get the integer back, not the `EmployeeType.PART_TIME` enum.

**Comment by onlined** [comment]
@tseaver Yes, but it would be comparable with the enum, and I think that's the most common case of using an enum.
```py
if employee_entity['type'] == EmployeeType.PART_TIME:
	# Do something here
	pass
```
